core-data/index.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * WordPress Dependencies
+ */
+import { registerStore } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import reducer from './reducer';
+import { getCategories } from './selectors';
+
+/**
+ * Module Constants
+ */
+const MODULE_KEY = 'core';
+
+const store = registerStore( MODULE_KEY, {
+	reducer,
+	selectors: {
+		getCategories,
+	},
+	resolvers: {
+		getCategories() {
+			wp.apiRequest( { path: '/wp/v2/categories' } ).then( ( categories ) => {
+				store.dispatch( {
```

Comment:
In this case we don't even need to specify an action creator as it seems it shouldn't be executed in other places. However, it might make sense to always create a corresponding action creator to make it easier to extend. If someone wants to override this resolver, they will need to have a simple way to dispatch it anyways.

Reply:
>However, it might make sense to always create a corresponding action creator to make it easier to extend.

I also think it sets a good precedent for those who learn by ~copying~ referencing core code.